<a href="https://colab.research.google.com/github/felixetec/programacao/blob/main/avalia%C3%A7%C3%A3oPA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

QUESTÃO 01 – Contagem Crescente (2 pontos)
---------------------------------------------------------

Elabore um programa que exiba na tela os números de 1 até 20,
um em cada linha.

Exemplo de saída:

1
2
3
...
20

In [2]:
for i in range(1,21):
  print(i)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20


QUESTÃO 02 – Soma dos Números (2 pontos)
---------------------------------------------------------

Solicite ao usuário um número inteiro positivo.

Utilizando um laço de repetição, calcule e exiba a soma de
todos os números de 1 até o número informado.

Exemplo:

Digite um número: 5

Resultado: 15

Pois:

# 1 + 2 + 3 + 4 + 5 = 15

In [60]:
soma = 0
n = int(input("Digite um número: "))
for i in range(n + 1):
  soma += i
  print(n + soma - n)

Digite um número: 20
0
1
3
6
10
15
21
28
36
45
55
66
78
91
105
120
136
153
171
190
210


# Solicite ao usuário um número inteiro e exiba sua tabuada
de 1 até 10.

Exemplo:

Digite um número: 7

7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
...
7 x 10 = 70

In [15]:
n = int(input(""))
for i in range(1, 11):
  print(f"{n} x {i} = {n * i}")


2
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20


#Solicite ao usuário um número inteiro positivo e calcule
seu fatorial utilizando um laço de repetição.

Exemplo:

Digite um número: 5

Fatorial = 120

Pois:

5! = 5 × 4 × 3 × 2 × 1

In [59]:
n = int(input("digite um número: "))

if n < 0:
    print("fatorial não existe para números negativos.")
elif n == 0:
    print("fatorial de 0 é 1.")
else:
    factorial = 1
    for i in range(1, n + 1):
        factorial *= i
    print(f"fatorial = {factorial}")

digite um número: 5
fatorial = 120


QUESTÃO 05 – Caixa Eletrônico (5 pontos)
---------------------------------------------------------

Um caixa eletrônico possui notas de:

R$ 100
R$ 50
R$ 20
R$ 10
R$ 5
R$ 2
R$ 1

Solicite ao usuário o valor do saque e informe a menor
quantidade possível de notas para realizar o pagamento.

Exemplo:

Digite o valor do saque: 287

Saída esperada:

2 nota(s) de R$100
1 nota(s) de R$50
1 nota(s) de R$20
1 nota(s) de R$10
1 nota(s) de R$5
1 nota(s) de R$2

Requisitos:
- Utilizar laços de repetição.
- Não utilizar listas.
- Exibir apenas as notas utilizadas.

In [67]:
saque = int(input("digite o valor que deseja sacar: "))
n = (100, 50, 20, 10, 5, 2, 1)
print('')
for nota in n:
  q = saque // nota
  if q > 0:
    print(f"{q} nota(s) de R${nota}")
    saque %= nota

digite o valor que deseja sacar: 253

2 nota(s) de R$100
1 nota(s) de R$50
1 nota(s) de R$2
1 nota(s) de R$1
